Add artifact to secure MariaDB installations on Linux

sjohner opened the pull request:
The artifact accepts parameter for new root password and is based on [Twitters implementation of mysql_secure_installation.sh](https://github.com/twitter/mysql/blob/master/scripts/mysql_secure_installation.sh)

It basically does the following:

- Set root password and make sure that nobody can access the server without a password
- Remove anonymous users
- Disallow root login remotely
- Remove test database and access to it
- Reload privilege tables

Tested with the following images:

- Debian 9 "Stretch"
- Ubuntu 16.04
- CentOS 7.4

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/azure-devtestlab?pullRequest=352) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/azure-devtestlab?pullRequest=352) before we can accept your contribution.<br/><br/>:x: sjohner [sign now](https://cla.opensource.microsoft.com/Azure/azure-devtestlab?pullRequest=352)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/azure-devtestlab?pullRequest=352) it.</sub>

leovms commented on Artifacts/linux-secure-mariadb/README.md:
```diff
@@ -0,0 +1,21 @@
+# Linux Secure MariaDB Artifact
+This Azure DevTest artifact allows the user to secure MariaDB installations already present on a Azure DevTest Lab VM (for example installed via Apt/Yum Package Manager Artifact). This artifact applies to any Linux distribution.
```
Azure DevTest Labs

leovms commented on Artifacts/linux-secure-mariadb/README.md:
```diff
@@ -0,0 +1,21 @@
+The script is intended for use in the Azure DevTest Labs artifact system, and the parameters for the artifact are fed directly
+into the script. However, you can run it from any bash shell using the following format:
+
+        bash> ./linux-secure-mariadb.sh [-p=password]  
```
This bash example doesn't match the syntax we're using in the `artifactfile.json` file. Can we in fact use both parameter formats (i.e. `-p=password` and `-p password`)?

leovms commented on Artifacts/linux-secure-mariadb/linux-secure-mariadb.sh:
```diff
@@ -0,0 +1,69 @@
+#
+# Usage: 
+#
+# linux-secure-mariadb.sh [-p=password]
```
Again, just pointing out this example does't match the `USAGE_STRING` defined just below.

sjohner commented:
Thanks for the feedback! Fixed error reporting and updated usage samples to be consitent. Also added checks to make sure mysql is available and maria db is running before making changes to the database.

sjohner commented:
Any update on this?
